charts/drax/templates/secret-grafana-slack-contactpoints.yaml

```
@@ -0,0 +1,34 @@
+{{- if $.Values.global.alerting.enabled -}}
+{{- if eq $.Values.global.alerting.slackToken "" -}}
+  {{- fail "Error: global.alerting.slackToken must be defined." -}}
+{{- end -}}
+{{- end -}}
+
+{{- $secretName := printf "%s-slack-auth" $.Release.Name -}}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ $secretName }}
+  annotations:
+    "helm.sh/resource-policy": keep
+type: Opaque
+data:
+  {{- $existingSecret := lookup "v1" "Secret" .Release.Namespace $secretName }}
+  {{- if $.Values.global.alerting.enabled -}}
+    {{- if and $existingSecret $existingSecret.data (hasKey $existingSecret.data "slack-token") -}}
+      {{- $slackToken := get ($existingSecret.data) "slack-token" | b64dec }}
+      {{- if eq $slackToken "" -}}
+        slack-token: {{ $.Values.global.alerting.slackToken | b64enc | quote }}
+      {{- else }}
+        slack-token: {{ get ($existingSecret.data) "slack-token" | quote }}
+      {{- end }}
+    {{- else }}
+      slack-token: {{ $.Values.global.alerting.slackToken | b64enc | quote }}
+    {{- end }}
+  {{- else }}
+      {{- if not (and $existingSecret $existingSecret.data (hasKey $existingSecret.data "slack-token")) -}}
+      slack-token: ""
+      {{- else }}
+      slack-token: {{ get ($existingSecret.data) "slack-token" | quote }}
+      {{- end }}
+  {{- end }}
```

I think that this is equivalent, but to be double checked (it mainly removes the need to check the existing secret data twice).

```suggestion
data:
  {{- $existingSecret := lookup "v1" "Secret" .Release.Namespace $secretName }}

  {{- $existingSlackToken := "" }}
  {{- if and $existingSecret $existingSecret.data (hasKey $existingSecret.data "slack-token") }}
    {{- $existingSlackToken = get ($existingSecret.data) "slack-token" | b64dec }}
  {{- end }}

  {{- $slackToken := $existingSlackToken }}
  {{- if and ((($.Values).global).alerting).enabled (not $slackToken) }}
    {{- $slackToken = $.Values.global.alerting.slackToken }}
  {{- end }}
  slack-token: {{ $slackToken | b64enc | quote }}

```